### PR TITLE
Make editor ui work inside egui windows.

### LINKED ISSUE #4: Passing 'Ui' to draw_graph_editor
Currently `draw_graph_editor` creates a CentralPanel to draw its stuff. In my case I would prefer if the the main application manages the container for for the graph_editor and just passes a `Ui` to `draw_graph_editor`. Is there a specific reason for the CentralPanel approach (like special requirement for the pan-zoom) ?

### PR DESCRIPTION
**Pull request by nodeSpace**
Proposal to fix #4
Note, this would be a breaking change and will require users to pass in an `&ui` when calling the `draw_graph_editor()`

### CODE REVIEW

**Comment by gmorenz** [comment]
I just started using this library and this was the first thing I wanted :+1: 

Two nits:

 - The example needs updating, just changing `self.state.draw_graph_editor(ctx, AllMyNodeTemplates)` to
 
   ```rust
   let graph_response =
      egui::CentralPanel::default().show(&ctx, |ui| {
          self.state.draw_graph_editor(ctx, ui,  AllMyNodeTemplates)
      }).inner;
   ```
   and maybe rewording the comment two lines up 
 
 - It seems like nodes are being drawn relative to the screen instead of the UI, i.e. using the code below, the nodes in the window don't move with the window when it is moved (but I haven't checked the code to confirm why this is happening)
 
   ```rust
   egui::Window::new(&editor.title).show(ctx, |ui| {
       editor.graph.draw_graph_editor(ctx, ui, AllNodeTemplates);
   });
   ```

**Comment by kkngsm** [comment]
Also, the `ctx` argument may not be necessary since `ctx` can be gotten by `ui.ctx()`.

**Comment by setzer22** [comment]
I cleared up the merge conflicts and apply some suggestions from this thread. Merging this, thank you all for the great work! :smile:
